internal/otelcollector/config/loggateway/config_builder.go

```diff
@@ -33,118 +32,345 @@ type BuildOptions struct {
+			return common.K8sAttributesProcessorConfig(opts.Enrichments)
+		},
+	)
+}
+
+func (b *Builder) addIstioNoiseFilterProcessor() buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("istio_noise_filter"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			return &common.IstioNoiseFilterProcessor{}
+		},
+	)
+}
+
+func (b *Builder) addDropIfInputSourceOTLPProcessor() buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("filter/drop-if-input-source-otlp"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			if logpipelineutils.IsOTLPInputEnabled(lp.Spec.Input) {
+				return nil // Skip this processor if OTLP input is enabled
+			}
+
+			return dropIfInputSourceOTLPProcessorConfig()
+		},
+	)
 }
 
-// addComponentsForLogPipeline enriches a Config (exporters, processors, etc.) with components for a given telemetryv1alpha1.LogPipeline.
-func (b *Builder) addComponentsForLogPipeline(ctx context.Context, pipeline *telemetryv1alpha1.LogPipeline, queueSize int) error {
-	b.addNamespaceFilter(pipeline)
-	b.addInputSourceFilters(pipeline)
-	b.addUserDefinedTransformProcessor(pipeline)
+func (b *Builder) addNamespaceFilterProcessor() buildComponentFunc {
+	return b.addProcessor(
+		formatNamespaceFilterID,
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			otlpInput := lp.Spec.Input.OTLP
+			if otlpInput == nil || otlpInput.Disabled || !shouldFilterByNamespace(otlpInput.Namespaces) {
+				return nil // No namespace filter needed
+			}
 
-	return b.addOTLPExporter(ctx, pipeline, queueSize)
+			return namespaceFilterProcessorConfig(otlpInput.Namespaces)
+		},
+	)
 }
 
-func (b *Builder) addNamespaceFilter(pipeline *telemetryv1alpha1.LogPipeline) {
-	otlpInput := pipeline.Spec.Input.OTLP
-	if otlpInput == nil || otlpInput.Disabled {
-		// no namespace filter needed
-		return
+func (b *Builder) addInsertClusterAttributesProcessor(opts BuildOptions) buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("resource/insert-cluster-attributes"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			return common.InsertClusterAttributesProcessorConfig(opts.ClusterName, opts.ClusterUID, opts.CloudProvider)
+		},
+	)
+}
+
+func (b *Builder) addServiceEnrichmentProcessor() buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("service_enrichment"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			return common.ResolveServiceNameConfig()
+		},
+	)
+}
+
+func (b *Builder) addDropKymaAttributesProcessor() buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("resource/drop-kyma-attributes"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			return common.DropKymaAttributesProcessorConfig()
+		},
+	)
+}
+
+func (b *Builder) addIstioEnrichmentProcessor(opts BuildOptions) buildComponentFunc {
+	return b.addProcessor(
+		staticComponentID("istio_enrichment"),
+		func(lp *telemetryv1alpha1.LogPipeline) any {
+			return istioEnrichmentProcessorConfig(opts)
```

Comment:
since `istioEnrichmentProcessorConfig` func is just a small return statement and only used once here, maybe we can just define it here?

Reply:
Yeah, makes sense